### PR TITLE
[#212] Fixes #212 - auth for harvest_job_create was broken.

### LINKED ISSUE #212: Trying to harvest you get error "[No module named auth.create])"
When a non-sysadmin tries to start a harvest manually you get the flash error:

    An error occurred: [No module named auth.create])

and it doesn't harvest.

Bug on master since Nov 20th when this PR was merged: https://github.com/ckan/ckanext-harvest/issues/111

### PR DESCRIPTION
**Pull request by davidread**
Issue #212

### CODE REVIEW

File: ckanext/harvest/logic/action/update.py [hunk 1]
```diff
@@ -462,16 +462,16 @@ def harvest_send_job_to_gather_queue(context, data_dict):
     :type id: string
     '''
     log.info('Send job to gather queue: %r', data_dict)
-    check_access('harvest_send_job_to_gather_queue', context, data_dict)
 
     job_id = logic.get_or_bust(data_dict, 'id')
+    job = toolkit.get_action('harvest_job_show')(
+        context, {'id': job_id})
+
+    check_access('harvest_send_job_to_gather_queue', context, job)
```
Comment (davidread) [inline ckanext/harvest/logic/action/update.py]: harvest_send_job_to_gather_queue needs to know the source_id, so bring the harvest_job_show call earlier and pass in the result.

File: ckanext/harvest/tests/test_action.py [hunk 2]
```diff
@@ -403,6 +403,44 @@ def test_harvest_source_create_twice_with_unique_url_and_config(self):
...
+            }
+        job = toolkit.get_action('harvest_job_create')(
+            {'user': site_user}, data_dict)
+
+        assert_equal(job['source_id'], source['id'])
+        assert_equal(job['status'], 'Running')
+        assert_equal(job['gather_started'], None)
+        assert_in('stats', job.keys())
+
+    def test_harvest_job_create_as_admin(self):
```
Comment (davidread) [inline ckanext/harvest/tests/test_action.py]: This was the failing test

**Comment by davidread** [comment]
Merging, as it's an important fix

File: ckanext/harvest/tests/test_action.py [hunk 3]
```diff
@@ -403,6 +403,44 @@ def test_harvest_source_create_twice_with_unique_url_and_config(self):
...
+            'source_id': source['id'],
+            'run': True
+            }
+        job = toolkit.get_action('harvest_job_create')(
+            {'user': user['name']}, data_dict)
+
+        assert_equal(job['source_id'], source['id'])
+        assert_equal(job['status'], 'Running')
+        assert_equal(job['gather_started'], None)
+        assert_in('stats', job.keys())
```
Comment (amercader) [inline ckanext/harvest/tests/test_action.py]: @davidread `assert_in` is not present on CKAN <= 2.2 so tests are failing on this version. Can you add a fallback like this one on core? https://github.com/ckan/ckan/blob/master/ckan/tests/helpers.py#L30:L35
Thanks

**Pull request merged**